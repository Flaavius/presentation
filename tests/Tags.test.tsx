import React from "react";
import { shallow } from "enzyme";

import { Tags, ITag } from "../src/Components/Tags";

const tagsTemplate: ITag[] = [
  {
    title: "This",
    url: "http://this",
  },
  {
    title: "is",
    url: "http://is",
  },
  {
    title: "a",
    url: "http://a",
  },
  {
    title: "test",
    url: "http://test",
  },
];

describe("Tags", () => {
  const tag = shallow(<Tags tags={tagsTemplate} />);
  it("passes props correctly", () => {
    expect(tag.props()).toMatchSnapshot();
  });
  it("Renders with appropriate tags", () => {
    expect(tag).toMatchSnapshot();
  });
});
