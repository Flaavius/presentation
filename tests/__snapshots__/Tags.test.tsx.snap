// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags Renders with appropriate tags 1`] = `
<div
  className="tags"
>
  <a
    href="http://this"
    key="0"
  >
     
    This
     
  </a>
  <a
    href="http://is"
    key="1"
  >
     
    is
     
  </a>
  <a
    href="http://a"
    key="2"
  >
     
    a
     
  </a>
  <a
    href="http://test"
    key="3"
  >
     
    test
     
  </a>
</div>
`;

exports[`Tags passes props correctly 1`] = `
Object {
  "children": Array [
    <a
      href="http://this"
    >
       
      This
       
    </a>,
    <a
      href="http://is"
    >
       
      is
       
    </a>,
    <a
      href="http://a"
    >
       
      a
       
    </a>,
    <a
      href="http://test"
    >
       
      test
       
    </a>,
  ],
  "className": "tags",
}
`;
